model/src/main/kotlin/utils/SimplePackageConfigurationProvider.kt

```diff
@@ -36,6 +36,10 @@ import org.ossreviewtoolkit.model.readValue
 open class SimplePackageConfigurationProvider(
     configurations: Collection<PackageConfiguration>
 ) : PackageConfigurationProvider {
+    companion object {
+        fun forDirectory(directory: File?): PackageConfigurationProvider =
```

Comment:
IIRC it was a deliberate design choice by @mnonnenmacher to remove this function in d10bd49466ba04afbf748397e2413e6872f38f2a.

Reply:
Thanks, I was aware of this choice. My commit here is removing the `createProvider()` function from `utils`.

I was in need of a replacement for this function and this was my best guess. It can also be used from `cli` module, as soon as I also remove the `--package-configuration-file` option there.

Reply:
It seems wrong to me to let `SimplePackageConfigurationProvider` create a `DirectoryPackageConfigurationProvider` instead of letting the latter do that itself. I guess I'd simply inline this implementation to the caller, also see my other comment.

Reply:
Another alternative would be to make `DirectoryPackageConfigurationProvider`'s `directory` parameter nullable and handle this as part of its constructor.